### DISCUSSION TITLE
Add support for self-hosted source control

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cloud#31] [Feature Requests & Enhancements]
I would be nice to add support for self-hosted versions of Github, Gitlab, and Bitbucket.

* https://about.gitlab.com/install/
* https://docs.github.com/en/enterprise-server@3.14/admin/installing-your-enterprise-server
* https://confluence.atlassian.com/bitbucketserver/bitbucket-installation-guide-867338382.html

**Comment by holmey** [comment]
This would be really great. We would need support for a self-hosted GitLab.
Another way would be a CLI tool capable of pushing the code to the Cloud project like in e.g. [Vercel](https://vercel.com/).
